### PR TITLE
refactor:firewall module

### PR DESCRIPTION
**Pull request by blackHatMonkey**
* Refactor the firewall module
* Add documentation
* Add type annotations

### CODE REVIEW

File: wifiphisher/common/firewall.py [hunk 1]
```diff
@@ -1,46 +1,60 @@
...
+            "--to-destination {}:{}".format(NETWORK_GW_IP, SSL_PORT)
+        ])
 
     def on_exit(self):
+        # type: () -> None
+        """Start the clean up."""
         self.clear_rules()
+
+
+def execute_commands(commands):
```
Comment (sophron) [inline wifiphisher/common/firewall.py]: This doesn't look like something that belongs here. Maybe create a `utils.py` file and put it there.

File: wifiphisher/common/firewall.py [hunk 2]
```diff
@@ -1,46 +1,60 @@
...
+                internal_interface, external_interface)
+        ])
 
     def clear_rules(self):
-        subprocess.call('iptables -F', shell=True)
-        subprocess.call('iptables -X', shell=True)
-        subprocess.call('iptables -t nat -F', shell=True)
-        subprocess.call('iptables -t nat -X', shell=True)
+        # type: () -> None
+        """Reset all the routing."""
```
Comment (sophron) [inline wifiphisher/common/firewall.py]: Change to: Clears all rules.

File: wifiphisher/common/firewall.py [hunk 3]
```diff
@@ -1,46 +1,60 @@
...
+                        unicode_literals)
+import logging
+from subprocess import (Popen, PIPE)
+from wifiphisher.common.constants import (NETWORK_GW_IP, SSL_PORT, PORT, DN)
 
 
 class Fw():
-    def __init__(self):
-        pass
+    """Handle iptable routing."""
```
Comment (sophron) [inline wifiphisher/common/firewall.py]: Change to: Handles all iptables operations.

File: wifiphisher/common/firewall.py [hunk 4]
```diff
@@ -1,46 +1,60 @@
...
-            ('iptables -t nat -A POSTROUTING -o %s -j MASQUERADE' %
-             (external_interface, )),
-            shell=True)
-
-        subprocess.call(
-            ('iptables -A FORWARD -i %s -o %s -j ACCEPT' %
-             (internal_interface, external_interface)),
-            shell=True)
+        # type: (str, str) -> None
+        """Create NAT in iptalbes."""
```
Comment (sophron) [inline wifiphisher/common/firewall.py]: Change to: Does NAT.

**Comment by blackHatMonkey** [comment]
@sophron I made the changes you requested.

Regarding the feature it should be easy to implement I just don't know the `iptable` commands. If you can post those I should be able to do (in a separate PR).

Let me know when you are done so I can squash the changes :smile:.

File: wifiphisher/common/firewall.py [hunk 5]
```diff
@@ -1,46 +1,56 @@
...
+        """Clear all rules."""
+        execute_commands([
+            "iptables -F", "iptables -X", "iptables -t nat -F",
+            "iptables -t nat -X"
+        ])
+
+    @staticmethod
+    def redirect_requests_localhost():
+        # type: () -> None
+        """Redirect all the request to local host.
```
Comment (sophron) [inline wifiphisher/common/firewall.py]: You can delete this line.

File: wifiphisher/common/firewall.py [hunk 6]
```diff
@@ -1,46 +1,56 @@
...
+
+        Redirects follwing request to localhost:
```
Comment (sophron) [inline wifiphisher/common/firewall.py]: Change to: Redirect the following requests to localhost:

**Comment by sophron** [comment]
I have already mentioned the commands in my previous post: `iptables-save` and `iptables-restore`. 

You will find many resources on how to use them on the Internet. Here's one: https://askubuntu.com/questions/66890/how-can-i-make-a-specific-set-of-iptables-rules-permanent

They are fairly simple.

File: wifiphisher/common/firewall.py (hunk 5 above)
Comment (blackHatMonkey) [reply wifiphisher/common/firewall.py]: due to [pep257](https://www.python.org/dev/peps/pep-0257/#id17) we need a summary line so we either have to keep the same or if you have another alternative I can change it to.

Comment (sophron) [reply wifiphisher/common/firewall.py]: Change it to: Redirect HTTP, HTTPS & DNS requests to localhost.

**Comment by blackHatMonkey** [comment]
This is ready to be merged :smile:.

**Comment by sophron** [comment]
Thank you.

**Pull request merged**